File: java/src/main/java/it/giannibombelli/wsc2026/payment/domain/events/PaymentDeadlineReached.java
package it.giannibombelli.wsc2026.payment.domain.events;

import it.giannibombelli.wsc2026.common.utils.Require;

import it.giannibombelli.wsc2026.payment.domain.payment.PaymentId;

public record PaymentDeadlineReached(
    PaymentId aggregateId
) implements PaymentEvent {

    public PaymentDeadlineReached {
        Require.requireArgument(aggregateId, "paymentId");
    }
}

File: typescript/src/payment/domain/events/paymentDeadlineReached.ts
// Evento emesso quando viene raggiunta la scadenza di 48h per un pagamento.

import type { Event } from '@/common/domain/model/event.js';
import type { PaymentId } from '../payment/paymentId.js';
import { requireArgument } from '@/common/utils/requireArgument.js';

export type PaymentDeadlineReached = Event<'PaymentDeadlineReached', PaymentId>;

export function paymentDeadlineReached(aggregateId: PaymentId): PaymentDeadlineReached {
  requireArgument(aggregateId, 'paymentId');
  return { kind: 'PaymentDeadlineReached', aggregateId };
}
